include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -195,6 +205,17 @@ typedef struct OrtAllocator {
   void*(ORT_API_CALL* Alloc)(struct OrtAllocator* this_, size_t size);
   void(ORT_API_CALL* Free)(struct OrtAllocator* this_, void* p);
   const struct OrtMemoryInfo*(ORT_API_CALL* Info)(const struct OrtAllocator* this_);
+
+  // Following functions are meant for arena based allocators and optional to implement
+  // if you're not implementing your allocator based on an arena.
+  // Reserve size bytes
+  void*(ORT_API_CALL* Reserve)(struct OrtAllocator* this_, size_t size);
```

Comment:
Required to satisfy these types of calls https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/core/framework/tensor_allocator_with_mem_pattern.h#L44